courses/views.py

```diff
@@ -106,3 +116,52 @@ def user_existence(request):
         return Response({"error": "Must provide a UID"}, status=400)
 
     return Response({"exists": EdxAuthor.objects.filter(edx_uid=uid).exists()})
+
+
+@api_view(['POST'])
+def create_ccx(request):
+    """
+    Supports creating a CCX on edx.
+    """
+    def upstream_error(msg=None):
+        """Generate logged errors for upstream response errors"""
+        err = {
+            'error': 'Unable to create course on upstream edx server. {}'.format(msg)
+        }
+        log.error(err['error'])
+        return Response(err, status=502)
+
+    for required in ('master_course_id', 'user_email', 'total_seats', 'display_name'):
+        if required not in request.POST:
+            return Response({
+                'error': 'You did not supply required POST argument {}'.format(required),
+            }, status=400)
+
+    master_course_id = request.POST['master_course_id']
+    course = get_object_or_404(Course, course_id=master_course_id)
+    access_token = get_access_token(course.edx_instance)
+    user_email = request.POST['user_email']
+
+    try:
+        resp = requests.post(
+            '{instance}/api/ccx/v0/ccx/'.format(instance=course.edx_instance.instance_url),
+            data={
+                'master_course_id': master_course_id,
+                'coach_email': user_email,
+                'max_students_allowed': request.POST['total_seats'],
+                'display_name': request.POST['display_name'],
+            },
+            headers={
+                'Authorization': 'Bearer {}'.format(access_token),
+            })
+    except RequestException as e:
+        return upstream_error(e)
+
+    if resp.status_code >= 300:
+        return upstream_error(resp.content)
```

Comment:
Can the teachers portal use the single `error_code` string and react accordingly? 
If not it is fine as it is, otherwise it is worth sending it in a specific field.